src/sql-parser/src/parser.rs

```diff
@@ -2629,14 +2629,22 @@ impl<'a> Parser<'a> {
         let if_not_exists = self.parse_if_not_exists()?;
         let name = self.parse_item_name()?;
 
-        // Webhook Source, which works differently than all other sources.
-        if self.peek_keywords(&[FROM, WEBHOOK]) {
-            return self.parse_create_webhook_source(name, if_not_exists);
-        }
-
         let (col_names, key_constraint) = self.parse_source_columns()?;
+
+        // For webhook sources we require `IN CLUSTER`, so track the position for an error message.
+        let cluster_err = self.expected(self.peek_pos(), "IN CLUSTER", self.peek_token());
         let in_cluster = self.parse_optional_in_cluster()?;
         self.expect_keyword(FROM)?;
+
+        // Webhook Source, which works differently than all other sources.
```

Comment:
Why do they need to work differently? Can webhooks be merged into the normal CreateSourceStatement struct?

Reply:
Admittedly this is something I keep bumping up against. I think webhooks could be merged into the `CreateSourceStatement`, but I'm not confident it would necessarily be a cleaner implementation than what we have currently. 

If it's okay with you I'd like to keep them separate for now, and as we make further changes (like this one) merge them together as it naturally works. If you would really like to see them merged though let me know and I'll work on that!

Reply:
We're intentionally not supporting the new `SIZE` option because we want to discourage users from creating clusters just for a single webhook source, since that cluster won't end up doing any work. The plan is once webhook sources actually run on `clusterd` then we'll support the `SIZE` option

Reply:
sgtm